python/cudf/cudf/core/dataframe.py

```diff
@@ -777,7 +777,16 @@ def __setitem__(self, arg, value):
                     self.insert(len(self._data), arg, value)
 
         elif isinstance(
-            arg, (list, np.ndarray, pd.Series, Series, Index, pd.Index)
+            arg,
+            (
+                list,
+                np.ndarray,
+                cupy.ndarray,
```

Comment:
Should we generally handle objects with `__cuda_array_interface__` here?

Reply:
Yup, we have generic utility `is_column_like` for this. I changed it to `can_convert_to_column` to also handle list-like inputs.